### PR TITLE
[BEAM-551] Fix handling of null ValueProviders in DisplayData

### PR DESCRIPTION
**Pull request by sammcveety**
R: @swegner

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/display/DisplayDataTest.java [hunk 1]
```diff
@@ -379,6 +379,7 @@ public void testAddIfNotNull() {
       public void populateDisplayData(Builder builder) {
         builder
             .addIfNotNull(DisplayData.item("nullString", (String) null))
+            .addIfNotNull(DisplayData.item("nullVPString", (ValueProvider<String>) null))
```
Comment (swegner) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/display/DisplayDataTest.java]: Should it also handle `addIfNotNull(DisplayData.item("nullVPValue", ValueProvider.of((String) null))` ?

**Comment by swegner** [comment]
LGTM.
R: @tgroh for committer approval and merge

**Comment by tgroh** [comment]
[INFO] Load of known null value in org.apache.beam.sdk.transforms.display.DisplayData.item(String, ValueProvider) [org.apache.beam.sdk.transforms.display.DisplayData] At DisplayData.java:[line 871]

It doesn't look like an actual bug; I think you may be able to get away with passing null explicitly.

**Comment by sammcveety** [comment]
Done

**Pull request closed without merging**